### PR TITLE
Add cluster support for redis extension

### PR DESCRIPTION
**Pull request by AnuGayan**

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java [hunk 1]
```diff
@@ -89,29 +97,55 @@
...
                         description = "password to connect to redis server",
                         type = {DataType.STRING}, optional = true,
-                        defaultValue = "will try to connect without authentication")
+                        defaultValue = "will try to connect without authentication"),
+                @Parameter(name = "cluster.mode",
+                        description = "This will decide the redis mode. if this is false, client will connect to a " +
+                                "single redis node.",
+                        type = {DataType.BOOL},
+                        defaultValue = "false"),
+                @Parameter(name = "nodes",
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java:94]: Cannot understand the difference between nodes and host, and why they have null as defaults ?

File: component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java [hunk 2]
```diff
@@ -89,29 +97,55 @@
...
+                                "table.name='fooTable',cluster.mode=false)" +
                                 "define table fooTable(time long, date String)",
-                        description = "above collection will create a redis table with the name fooTable"
+                        description = "Above example will create a redis table with the name fooTable and work on a" +
+                                "single redis node."
+                ),
+                @Example(
+                        syntax = "@Store(type='redis', table.name='SweetProductionTable', " +
+                                "nodes='localhost:30001,localhost:30002,localhost:30003', cluster.mode='true')\n" +
+                                "@primaryKey('symbol')\n" +
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java:113]: Why not use nodes for both clusters and single node?

**Review by suhothayan — CHANGES_REQUESTED** [review]

File: component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java (hunk 1 above)
Comment (AnuGayan) [reply component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java:94]: In the single node implementation, we already has the 'host' and 'port' implementation. In the cluster mode, we need to have host and port for for all the master nodes, so thats why i have introduced a new new params call nodes. In a single node scenario, nodes will be null. thats why I provided null as default. In clustered scenario we will throw an error is nodes are not defined.

File: component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java (hunk 2 above)
Comment (AnuGayan) [reply component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java:113]: if we changed the host into nodes It won't backward compatible..

File: component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java (hunk 1 above)
Comment (suhothayan) [reply component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java:94]: Having different type input for each type is not a good practice 
Suggestion is to use only "nodes" and for the single node case nodes will have only one entry.

File: component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java (hunk 2 above)
Comment (suhothayan) [reply component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java:113]: you can internally support host and port, but remove it from the documentation. 
and also support the having one entry in node case.

Comment (AnuGayan) [reply component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java:113]: Done the requested fix. nodes can be used in cluster mode as well and host, port also there in implementation.

**Review by suhothayan — APPROVED** [review]

**Pull request merged**